app/src/main/java/org/mozilla/fenix/search/SearchDialogFragment.kt

```diff
@@ -155,6 +156,15 @@ class SearchDialogFragment : AppCompatDialogFragment(), UserInteractionHandler {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setStyle(STYLE_NO_TITLE, R.style.SearchDialogStyle)
+
+        startForResult = registerForActivityResult { result ->
+            val intent = result.data
+            intent?.getStringArrayListExtra(RecognizerIntent.EXTRA_RESULTS)?.first()?.also {
```

Comment:
Looks like we could omit the parameter:
`result.data?.getStringArrayListExtra(RecognizerIntent.EXTRA_RESULTS)?.first()?.also {`

Less code, happier us!

Reply:
Sure, I'll remove `val intent`